<launch>
    <!-- 仿真world -->
    <arg name="gazebo_enable" default="true"/>
	<arg name="world" default="$(find prometheus_gazebo)/gazebo_worlds/planning_worlds/drl_world_data_collection.world"/>
    <!-- Rviz config -->
    <arg name="rviz_enable" default="true"/>
    <arg name="rivz_config" default="$(find prometheus_simulator_utils)/launch/map_generator.rviz"/>

    <!-- 生成随机地图，并发布全局、局部点云 -->
    <node pkg="prometheus_simulator_utils" name="map_generator_node" type="map_generator_node" output="screen">
        <param name="swarm_num" value="1" type="int"/>
        <param name="map_name" value="motion_planning" type="string"/>
    </node>

    <!-- 启动Gazebo -->
    <group if="$(arg gazebo_enable)">
        <include file="$(find gazebo_ros)/launch/empty_world.launch">
            <arg name="world_name" value="$(arg world)"/>
        </include>
    </group>

	<!-- 启动Rviz-->
	<group if="$(arg rviz_enable)">
        <node type="rviz" name="rviz" pkg="rviz" args="-d $(arg rivz_config)"/>
    </group>
</launch>
